Mage/src/main/java/mage/abilities/effects/common/combat/ReselectDefenderTargetEffect.java

```diff
@@ -0,0 +1,118 @@
+package mage.abilities.effects.common.combat;
+
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import mage.abilities.Ability;
+import mage.abilities.Mode;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.combat.CombatGroup;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetDefender;
+
+public class ReselectDefenderTargetEffect extends OneShotEffect {
```

Comment:
After further thought I do prefer ReselectDefenderAttackedByTargetEffect.

Also, add references to the CR 508.7